src/game/battle.rs

```diff
@@ -60,20 +60,26 @@ fn attack(
     attacker: &Character,
     receiver: &mut Character,
     random: &dyn Randomizer,
-) -> (Attack, i32) {
+) -> (Attack, i32, Option<Condition>) {
     if random.is_miss(attacker.speed, receiver.speed) {
-        (Attack::Miss, 0)
+        (Attack::Miss, 0, None)
     } else {
         let damage = random.damage(attacker.damage(receiver));
         let xp = attacker.xp_gained(receiver, damage);
+        // Randomly get condition for now. It may later be consequence of the
+        // attacker's feasiblity ratio to produce a certain condition and receiver's to get it
+        let condition = random.condition(attacker.produces_condition(receiver));
+        if condition.is_some() {
+            receiver.receive_condition(condition);
+        }
 
         if random.is_critical() {
             let damage = damage * 2;
             receiver.receive_damage(damage);
-            (Attack::Critical(damage), xp)
+            (Attack::Critical(damage), xp, condition)
```

Comment:
I would consider passing the status effect inside the Attack enum. like a new variant Attack::Effect(Condition) or if you think it can be regular/critical + effect, Attack::Regular(i32, Condition) and Attack::Critical(i32, Condition)

Reply:
For a first version we'll have a single Effect. So I'll take a look at making it a type of Attack